### PR TITLE
Allow audit syscall rules remediations to group the syscalls

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

- Improve and extend rule skeleton matching with more explicit rule options for `action`, `list`, `arch`, `auid` and other filters.
- Make explicit the syscalls that can through the `syscall_groupings` parameter.
  - When `syscall_groupings` is empty, it will always add a new rule if not found.
- Make they key to use more explicit, instead of implicit through  'group'.
- [X] Update the Ansible remediations (8af4ced71baa5794bfa9be2cfcf9a9519066e597)
- [ ] ~Add tests for `fix_audit_syscall_rule`~ I think the unit tests are not suited for this function. The function directly edits the rule files in `/etc/audit` and would be better run on a VM. If will need a VM anyway, I'd rather check and improve the test scenarios if needed.
- [X] Improve/adapt test scenarios (I believe the current test scenarios served pretty well to find bugs).

#### Rationale:

- The function actually separated the syscalls into individual lines.

### CODE REVIEW

**Comment by yuumasato** [comment]
Given the '-F auid=0` changes in #6910, I'm considering bringing the input to the Bash template directly to the 'rule.yml'.

**Comment by yuumasato** [comment]
I think I'll rebase this PR on top of #6910 to take advantage of the rule selections in the profile.

**Comment by yuumasato** [comment]
I think the Bash remediations are in good shape and good for review.

**Review by JAORMX — COMMENTED** [review]
Any chance this could be added to the kubernetes templates as well?

**Comment by JAORMX** [comment]
@yuumasato some of them are using straight up MachineConfigs... while we could now replace those entries with jinja to show the actual audit rule. I could help with that if you want.

File: shared/templates/audit_rules_dac_modification/ansible.template [hunk 1]
```diff
@@ -7,95 +7,81 @@
...
-      - "{{ find_{{{ ATTR }}}.files | map(attribute='path') | list | first }}"
-  when: find_{{{ ATTR }}}.matched is defined and find_{{{ ATTR }}}.matched > 0
-
-- name: Inserts/replaces the {{{ ATTR }}} rule in rules.d when on x86
-  lineinfile:
-    path: "{{ all_files[0] }}"
-    line: "-a always,exit -F arch=b32 -S {{{ ATTR }}} -F auid>={{{ auid }}} -F auid!=unset -F key=perm_mod"
-    create: yes
-
+- name: Perform remediattion of Audit rules for {{{ ATTR | join(", ") }}} for x86 platform
```
Comment (ggbecker) [inline shared/templates/audit_rules_dac_modification/ansible.template:14]:
```suggestion
- name: Perform remediation of Audit rules for {{{ ATTR | join(", ") }}} for x86 platform
```

**Comment by yuumasato** [comment]
> Any chance this could be added to the kubernetes templates as well?

Do you mean to add the grouping capability to the kubernetes remediations?
I don't know how that could be done, but we can discuss it if you have any idea.

> @yuumasato some of them are using straight up MachineConfigs... while we could now replace those entries with jinja to show the actual audit rule. I could help with that if you want.

@JAORMX You mean with `url_encode()`?

**Comment by JAORMX** [comment]
@yuumasato and I talked, and there's nothing to do for the kubernetes remediations as of now.

File: shared/macros-ansible.jinja [hunk 2]
```diff
@@ -385,135 +385,160 @@ The macro requires following parameters:
...
-    patterns: "*.rules"
-  register: audit_syscalls_found_{{{ arch }}}_rules_d
-  loop: "{{ syscalls }}"
+- name: Declare syscalls found per file
+  set_fact: syscalls_per_file="{{ syscalls_per_file | combine( {item.files[0].path :[item.item] + syscalls_per_file.get(item.files[0].path, []) } ) }}"
+  loop: "{{ find_command.results | selectattr('matched') | list}}"
 
-- name: Get number of matched syscalls for architecture {{{ arch }}} in /etc/audit/rules.d/
-  set_fact: audit_syscalls_matched_{{{ arch }}}_rules_d="{{ audit_syscalls_found_{{{ arch }}}_rules_d.results|sum(attribute='matched')|int }}"
+- name: Declare files where syscalls where found
```
Comment (ggbecker) [inline shared/macros-ansible.jinja:434]:
```suggestion
- name: Declare files where syscalls were found
```

**Review by ggbecker — APPROVED** [review]
LGTM. Thank you for these changes. It was a really nice improvement to the project.

**Pull request merged**